src/simple_merkle/proof.rs

```diff
@@ -30,19 +31,28 @@ where
         root: &M::Output,
         leaf_hashes: &[M::Output],
     ) -> Result<(), RangeProofError> {
-        let tree = MerkleTree::<MemDb<M::Output>, M>::new();
+        let tree = MerkleTree::<NoopDb, M>::new();
         let mut siblings = self.siblings.iter().collect();
 
-        tree.check_range_proof(root, leaf_hashes, &mut siblings, self.start_idx as usize)
+        if leaf_hashes.len() != (self.end - self.start) as usize {
```

Comment:
`start` and `end` are public, so we don't know if `end` is indeed bigger than `start`. add a check here.

maybe an clean solution is to implement `fn len(&self) -> Result<u32>`. it doesn't need to be public api.

Reply:
done